src/sagemaker/session.py

```diff
@@ -343,34 +343,35 @@ def default_bucket(self):
             default_bucket = "sagemaker-{}-{}".format(region, account)
 
         s3 = self.boto_session.resource("s3")
-        try:
-            # 'us-east-1' cannot be specified because it is the default region:
-            # https://github.com/boto/boto3/issues/125
-            if region == "us-east-1":
-                s3.create_bucket(Bucket=default_bucket)
-            else:
-                s3.create_bucket(
-                    Bucket=default_bucket, CreateBucketConfiguration={"LocationConstraint": region}
-                )
+        bucket = self.boto_session.resource("s3").Bucket(name=default_bucket)
```

Comment:
`region_name` needs to be set in ll. 345-346

Reply:
Good catch!